lighthouse-cli/test/smokehouse/run-smoke.js

```diff
@@ -0,0 +1,191 @@
+ */
+'use strict';
+
+/* eslint-disable no-console */
+const {promisify} = require('util');
+const execAsync = promisify(require('child_process').exec);
+
+const {server, serverForOffline} = require('../fixtures/static-server');
+const log = require('lighthouse-logger');
+
+const purpleify = str => `${log.purple}${str}${log.reset}`;
+const smokehouseDir = 'lighthouse-cli/test/smokehouse/';
+
+/**
+ * @typedef {object} SmoketestDfn
+ * @property {string} id
+ * @property {string} expectations
+ * @property {string} config
+ * @property {string | undefined} batch
+ */
+
+/** @type {Array<SmoketestDfn>} */
+const SMOKETESTS = [{
+  id: 'ally',
+  config: smokehouseDir + 'a11y/a11y-config.js',
+  expectations: 'a11y/expectations.js',
+}, {
+  id: 'pwa',
+  expectations: smokehouseDir + 'pwa-expectations.js',
+  config: smokehouseDir + 'pwa-config.js',
+}, {
+  id: 'pwa2',
+  expectations: smokehouseDir + 'pwa2-expectations.js',
+  config: smokehouseDir + 'pwa-config.js',
+}, {
+  id: 'dbw',
+  expectations: 'dobetterweb/dbw-expectations.js',
+  config: smokehouseDir + 'dbw-config.js',
+}, {
+  id: 'redirects',
+  expectations: 'redirects/expectations.js',
+  config: smokehouseDir + 'redirects-config.js',
+}, {
+  id: 'seo',
+  expectations: 'seo/expectations.js',
+  config: smokehouseDir + 'seo-config.js',
+}, {
+  id: 'offline',
+  expectations: 'offline-local/offline-expectations.js',
+  config: smokehouseDir + 'offline-config.js',
+}, {
+  id: 'byte',
+  expectations: 'byte-efficiency/expectations.js',
+  config: smokehouseDir + 'byte-config.js',
+  batch: 'perf',
+}, {
+  id: 'perf',
+  expectations: 'perf/expectations.js',
+  config: 'lighthouse-core/config/perf-config.js',
+  batch: 'perf',
+}, {
+  id: 'ttci',
+  expectations: 'tricky-ttci/expectations.js',
+  config: 'lighthouse-core/config/default-config.js',
+}];
+
+/**
+ * Display smokehouse output from child process
+ * @param {{id: string, process: NodeJS.Process} || {id: string, error: Error & {stdout : NodeJS.WriteStream, stderr: NodeJS.WriteStream}}} result
+ */
+function displaySmokehouseOutput(result) {
+  console.log(`\n${purpleify(result.id)} smoketest results:`);
+  if (result.error) {
+    console.log(result.error.message);
+    process.stdout.write(result.error.stdout);
+    process.stderr.write(result.error.stderr);
+  } else {
+    process.stdout.write(result.process.stdout);
+    process.stderr.write(result.process.stderr);
+  }
+  console.timeEnd(`smoketest-${result.id}`);
+  console.log(`${purpleify(result.id)} smoketest complete. \n`);
+}
+
+/**
+ * Run smokehouse in child processes for selected smoketests
+ * Display output from each as soon as they finish, but resolve function when ALL are complete
+ * @param {Array<SmoketestDfn>} smokes
+ */
+async function runSmokehouse(smokes) {
+  const cmdPromises = [];
+  for (const {id, expectations, config} of smokes) {
+    console.log(`${purpleify(id)} smoketest starting…`);
```

Comment:
for some reason, just for the perf batch, Travis does 

![screen shot 2018-04-10 at 16 36 19](https://user-images.githubusercontent.com/316891/38588984-6d6a51fa-3cdd-11e8-877d-93043d0cbeb9.png)

not sure why :)